lib/ansible/module_utils/facts.py

```diff
@@ -2999,8 +2999,13 @@ def __new__(cls, *arguments, **keyword):
             return super(cls, subclass).__new__(subclass, *arguments, **keyword)
 
     def populate(self):
+        self.get_virtual_facts()
         return self.facts
 
+    def get_virtual_facts(self):
+        self.facts['virtualization_type'] = ''
+        self.facts['virtualization_role'] = ''
```

Comment:
> bcoca> alikins: in most cases we don't add keys when they are not detected, not just virtualization (which we also need to change into lists instead of fixed values)
> <bcoca> ^ we either change the behaviour to always populate all keys or keep as is, we should not change just for one set of keys
> <bcoca> it would be nice to be consistent one way or the other

Going by that, I'd suggest the base get_virtual_facts is

``` python
def get_virtual_facts(self):
    pass
```

But as you mention, that would change behavior. So pr as is LGTM to me, we can change the behavior later.